error.go

```diff
@@ -0,0 +1,21 @@
+package oidc
+
+import "errors"
+
+var (
+	ErrIssuerInvalid = errors.New("issuer does not match")
+)
+
+type Error struct {
+	ErrorType   string `json:"error,omitempty"`
+	Description string `json:"error_description,omitempty"`
+	State       string `json:"state,omitempty"`
+}
+
+func (e Error) Error() string {
+	message := "ErrorType=" + string(e.ErrorType)
```

Comment:
Maybe we shouldn't use uppercase to be consistent with other error messages?

Reply:
Generally speaking Go errors begin in lower case, true. What I am expecting here is that these types of structured errors will be wrapped with more context, for example:

```
err = httpRequest(...)
if err != nil {
  return nil, fmt.Errorf("failed to call token endpoint: %w", err)
}
```

The idea is that this could yield `failed to call token endpoint: ErrorType=invalid_client`, for example.